src/Symfony/Component/Form/ButtonBuilder.php

```diff
@@ -22,6 +25,11 @@
  */
 class ButtonBuilder implements \IteratorAggregate, FormBuilderInterface
 {
+    private const UNSUPPORTED_FORM_EVENTS = [
+        FormEvents::PRE_SET_DATA => [FormEvents::POST_SET_DATA],
```

Comment:
I just don't know if we can prevent userland from using our events?

I chose to not dispatch useless events. However, if userland dispatch them manually, it can still be useful to be able to listen to them. But I guess this would be a weird practice 😕 Personally, I just prefer to forbid them.